tests/cpp/test_combine_mul_sum.cpp

```diff
@@ -269,6 +271,344 @@ TEST_F(CombineMulSumAsMmaTest, UseMatmulScheduler) {
   }
 }
 
+using MatmulNodeTranslationTestParams =
+    std::tuple<int64_t, int64_t, bool, bool, bool, ScheduleHeuristic>;
+using MatmulNodeTranslationTest =
+    NVFuserFixtureParamTest<MatmulNodeTranslationTestParams>;
+
+// Test that a simple matmul op fusion is picked up by the appropriate scheduler
+// and the translation to MmaOp is performed properly.
+TEST_P(MatmulNodeTranslationTest, AutomaticSchedulerMatmulNode) {
+  const int64_t A_dim = std::get<0>(GetParam());
+  const int64_t B_dim = std::get<1>(GetParam());
+  const bool enable_fusion = std::get<2>(GetParam());
+  const bool transpose_a_alloc = std::get<3>(GetParam());
+  const bool expect_segmented = std::get<4>(GetParam());
+  const ScheduleHeuristic expected_heuristic = std::get<5>(GetParam());
+
+  // CombineMulSumAsMmaTest disabled MatmulExprEval, but we need it enabled
+  DisableOptionsGuard dog;
+  DisableOptionsGuard::getCurOptions().unset(DisableOption::MatmulExprEval);
+
+  EnableOptionsGuard eog;
+  if (enable_fusion) {
+    EnableOptionsGuard::getCurOptions().set(EnableOption::FuseMatmul);
+  } else {
+    EnableOptionsGuard::getCurOptions().unset(EnableOption::FuseMatmul);
```

Comment:
Do we need to unset it? The options guard should reset the value after each test, right?

Reply:
See other responses. You're correct but this makes it explicit and also overrides env vars that might be present.